app/src/app/components/routes/ProtectedRoute.tsx

```diff
@@ -18,15 +19,29 @@ export default function ProtectedRoute() {
     useEffect(() => {
         const user = getCurrentUser();
 
-        if (authConfig.enableAuth) {
-            if (!isAuthenticated && !user?.token) {
+        if (!authConfig.enableAuth) {
+            return;
+        }
+
+        if (!isAuthenticated && !user?.token) {
+            navigate("/login");
+            return;
+        }
+
+        if (user && user.token) {
+            const token = user.token;
+
+            if (isExpired(token)) {
                 navigate("/login");
-                return;
             }
 
-            if (user && user.token) {
-                dispatch(saveUser(user));
-            }
+            const decodedToken = decodeToken(token) as PackitDecodedToken;
+
+            const newUser = (decodedToken && decodedToken.email)
+                ? {...user, email: decodedToken.email}
```

Comment:
Could not test this line, not sure why the local storage does not work for object in test.